inc/entityinjection.class.php

```diff
@@ -87,27 +87,62 @@ function addOrUpdateObject($values = [], $options = []) {
    }
 
 
+   public static function getRootEntityName() {
+      $root = new Entity();
+      $root->getFromDb(0);
+
+      return $root->fields['name'];
+   }
+
+
     /**
     * @param $input     array
     * @param $add                (true by default)
     * @param $rights    array
    **/
    function customimport($input = [], $add = true, $rights = []) {
-
       if (!isset($input['completename']) || empty($input['completename'])) {
          return -1;
       }
 
+      $em = new Entity();
+
+      // Search for exisiting entity
+      $search = $input['completename'];
+
+      // Check if search start by root entity
+      $root = self::getRootEntityName();
+      if (strpos($search, $root) !== 0) {
+         $search = "$root > $search";
+      }
```

Comment:
I am not confident with this.
Complete name always contains the root entity name at start, so, IMHO, this piece of code tries to handle a case that should not exists.

Reply:
I forgot to specify this in the issues description but before these changes you had to use the completename without the root :
* If you imported "Entity A" it would create "Root entity > Entity A".
* If you imported "Root entity > Entity A" it would create "Root entity > Root entity > Entity A".

When I enabled the update I realized that it worked the opposite way (this time you needed the full completename to update the correct entity).

In the end I decided to allow both "full completename" and "rootless completename" name to work for creation and update.
This allow the more logical way to work (full completename) while still supporting the previous "rootless completename" version.

Reply:
OK, it make more sense.